R/zchunk_L2062.ag_Fert_irr_mgmt.R

```diff
@@ -30,74 +34,140 @@ module_aglu_L2062.ag_Fert_irr_mgmt_DISABLED <- function(command, ...) {
     all_data <- list(...)[[1]]
 
     # Load required inputs
-    L2061.AgCoef_Fert_ag_irr <- get_data(all_data, "L2061.AgCoef_Fert_ag_irr")
-    L2061.AgCoef_Fert_bio_irr <- get_data(all_data, "L2061.AgCoef_Fert_bio_irr")
-    L2061.AgCost_ag_irr_adj <- get_data(all_data, "L2061.AgCost_ag_irr_adj")
-    L2061.AgCost_bio_irr_adj <- get_data(all_data, "L2061.AgCost_bio_irr_adj")
-
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    GCAM_region_names <- get_data(all_data, "common/GCAM_region_names")
+    basin_to_country_mapping <- get_data(all_data, "water/basin_to_country_mapping")
+    A_Fodderbio_chars <- get_data(all_data, "aglu/A_Fodderbio_chars")
+    L142.ag_Fert_IO_R_C_Y_GLU <- get_data(all_data, "temp-data-inject/L142.ag_Fert_IO_R_C_Y_GLU")
+    L2052.AgCost_ag_irr_mgmt <- get_data(all_data, "temp-data-inject/L2052.AgCost_ag_irr_mgmt")
+    L2052.AgCost_bio_irr_mgmt <- get_data(all_data, "temp-data-inject/L2052.AgCost_bio_irr_mgmt")
+
+    # TEMPORARY: tidying until chunk LB142 complete
+    L142.ag_Fert_IO_R_C_Y_GLU %>%
+      gather(year, value, -GCAM_region_ID, -GCAM_commodity, -GLU) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L142.ag_Fert_IO_R_C_Y_GLU
+
+    # Process Fertilizer Coefficients: Copy coefficients to all four technologies (irr/rfd + hi/lo)
+    L142.ag_Fert_IO_R_C_Y_GLU %>%
+      filter(year %in% BASE_YEARS) %>%
+      left_join(GCAM_region_names, by="GCAM_region_ID") %>%
+      left_join(basin_to_country_mapping[ c("GLU_code", "GLU_name")], by=c("GLU" = "GLU_code")) %>%
+
+      # Copy coefficients to all four technologies
+      repeat_add_columns(tibble::tibble(IRR_RFD = c("IRR", "RFD")) ) %>%
+      repeat_add_columns(tibble::tibble(MGMT = c("hi", "lo")) ) %>%
```

Comment:
It doesn't actually. I have no idea why, but the data.frame ends up half as long as it should be.